src/Service.Tests/Unittests/SqlMetadataProviderUnitTests.cs

```diff
@@ -339,5 +340,135 @@ public void ValidateGraphQLReservedNaming_DatabaseColumns(string dbColumnName, s
+
+            // Validate that when custom source.fields/target.fields defined in the config for a relationship of cardinality N:1
+            // between Review - Book is the same as the FK constraint from Review -> Book,
+            // we successfully determine at the startup, that Review is the referencing entity.
+            ValidateReferencingEntitiesForRelationship("Review", "Book", new List<string>() { "Review" });
+
+            // Validate that when custom source.fields/target.fields defined in the config for a relationship of cardinality 1:N
+            // between Book - Review is the same as the FK constraint from Review -> Book,
+            // we successfully determine at the startup, that Review is the referencing entity.
+            ValidateReferencingEntitiesForRelationship("Book", "Review", new List<string>() { "Review" });
+
+            // Validate that when custom source.fields/target.fields defined in the config for a relationship of cardinality 1:1
+            // between Stock - stocks_price is the same as the FK constraint from stocks_price -> Stock,
+            // we successfully determine at the startup, that stocks_price is the referencing entity.
+            ValidateReferencingEntitiesForRelationship("Stock", "stocks_price", new List<string>() { "stocks_price" });
+
+            // Validate that when no custom source.fields/target.fields are defined in the config for a relationship of cardinality N:1
+            // between Book - Publisher and an FK constraint exists from Book->Publisher, we successfully determine at the startup,
+            // that Book is the referencing entity.
+            ValidateReferencingEntitiesForRelationship("Book", "Publisher", new List<string>() { "Book" });
+
+            // Validate that when no custom source.fields/target.fields are defined in the config for a relationship of cardinality 1:N
+            // between Publisher - Book and an FK constraint exists from Book->Publisher, we successfully determine at the startup,
+            // that Book is the referencing entity.
+            ValidateReferencingEntitiesForRelationship("Publisher", "Book", new List<string>() { "Book" });
+
+            // Validate that when no custom source.fields/target.fields are defined in the config for a relationship of cardinality 1:1
+            // between Book - BookWebsitePlacement and an FK constraint exists from BookWebsitePlacement->Book,
+            // we successfully determine at the startup, that BookWebsitePlacement is the referencing entity.
+            ValidateReferencingEntitiesForRelationship("Book", "BookWebsitePlacement", new List<string>() { "BookWebsitePlacement" });
+        }
+
+        private static void ValidateReferencingEntitiesForRelationship(
+            string sourceEntityName,
+            string targetEntityName,
+            List<string> referencingEntityNames)
```

Comment:
nit: Becomes apparent when reading the test logic, but re-naming would make the logic easier to understand
```suggestion
            List<string> expectedReferencingEntityNames)
```